### PR TITLE
Making rails application layout the default for preview index views

### LINKED ISSUE #289: Preview index actions use application layout instead of rails application layout
It seems that `/rails/view_components` page is rendered with "application" layout instead of "rails/application" layout. It is a problem because `RailsApplicationController` is missing methods implemented in `ApplicationController` and potentially used in the "application" layout. I checked the difference in implementation with mails preview and I can't really spot the difference.

### PR DESCRIPTION
**Pull request by juanmanuelramallo**
Resolves #289 

### Summary

- Updated view_components_controller to only prepend application's view paths when rendering a given preview
- Preview index views will use rails' default application layout
- Added integration tests for this new behaviour

image

image

image

### CODE REVIEW

**Comment by jonspalmer** [comment]
I think I understand how this fix works but it's really opaque. Is the fundamental problem that when we add the Rails app/view then there are two layouts called "application" and the Rails app version gets chosen over the Rails default?

**Comment by juanmanuelramallo** [comment]
What do you suggest @jonspalmer ?

**Comment by tchak** [comment]
No matter what is decided about preview controller inheritance I think listing layout/styles should be from "rails/application". So I am :+1: for this fix. I don't know enough about rails internals (and especially view layer) to know if there is a more elegant fix...

**Comment by jonspalmer** [comment]
Yeah I see the tests. My question is do we add a 4th test that makes three `get` requests inside the same test case. Make sure we're not relying on test order execution or some set of side effects we're not catching?

**Comment by juanmanuelramallo** [comment]
Definitely, good call!

File: test/view_component/integration_test.rb [hunk 1]
```diff
@@ -186,6 +185,35 @@ class IntegrationTest < ActionDispatch::IntegrationTest
     assert_select("div", "hello,world!")
   end
 
+  test "test preview renders application's layout by default" do
+    get "/rails/view_components/preview_component/default"
+
+    assert_includes response.body, "<title>ViewComponent - Test</title>"
```
Comment (joelhawksley) [inline test/view_component/integration_test.rb:191]: Can we use `assert_select` for these tests?

File: app/controllers/rails/view_components_controller.rb [hunk 2]
```diff
@@ -4,7 +4,6 @@
 
 class Rails::ViewComponentsController < Rails::ApplicationController # :nodoc:
   prepend_view_path File.expand_path("../../../lib/railties/lib/rails/templates/rails", __dir__)
-  prepend_view_path "#{Rails.root}/app/views/" if defined?(Rails.root)
```
Comment (joelhawksley) [inline app/controllers/rails/view_components_controller.rb:7]: Do we need a CHANGELOG for this?

Comment (juanmanuelramallo) [reply app/controllers/rails/view_components_controller.rb:7]: I'll add an entry

Comment (juanmanuelramallo) [reply app/controllers/rails/view_components_controller.rb:7]: Will this work https://github.com/github/view_component/pull/296/commits/c1a2a949f521612611bad1aebf06da32908756db ?

**Pull request merged**